### DISCUSSION TITLE
The prettier issue...

### DISCUSSION THREAD

**Discussion by Kernix13** [jdwilkin4/contrast-ratio-repo#33] [General]
@jdwilkin4 I assume you will be installing all the packages and we just need to pull the changes down. The package.json scripts commands are something I've been wanting to learn, mostly when using Webpack. Can you recommend any resources on how to write those scripts based on the packages being used?

**Comment by jdwilkin4** [comment]
Here is the WIP PR for the prettier issue 
https://github.com/DBenMoshe/contrast-ratio-repo/pull/31

The dependencies that will be added are `prettier`, `husky`, `lint-staged` and `browser-sync`
I will add to the readme what all of these packages are and instructions on how to install the dependencies. 

After the PR is merged, you will pull down those changes.
Run `npm install`

Every time you work on a new issue and push changes to GitHub, the prettier precommit hook will fire and automatically check to make sure that your code is formatted. If not, it will automatically format it for you. 

> Can you recommend any resources on how to write those scripts based on the packages being used?
Some of the packages will automatically add a script.

Husky added the ` "prepare": "husky install"` script

I added a ` "start": "browser-sync start --server --files .",` and found that in the documentation. 
When you run `npm start`, it will start the local server and use hot reloading when new changes are made.
Now people won't be dependent on installing local server extensions like Live Server on their IDE's and code editor. 
They can just run that command it will start the local server. 
You can still use the Live server extension, but now it will just be optional. 

But for writing script commands, always check with documentation first.
You can also study large open source codebases.
If you inspect their codebases, you can study their package.json files and see what kinds of scripts they wrote.

**Reply by AndreyDmitriev39r** [reply]
I am so new to this stuff.
So when we pull down all this, I will need to run `npm install` once? In bash(in the project folder)?
And then I will need to run `npm start` when working on project?

**Reply by jdwilkin4** [reply]
Yes, when you pull down the changes you will need to run `npm install` in the command line for the root of the project folder. 
You will only need to install the dependencies once. 

> And then I will need to run npm start when working on project?

If you want to keep using live server, or whatever local server extension you were using before, that is fine.

The goal of adding the `npm start` command was just an alternative option to starting a local server.
It is also a standard command to use for frontend library projects.
